gst_pipes_node:
  ros__parameters:


    gst_plugin_paths:
      - 'install/gst_bridge/lib/gst_bridge/'
    # XXX use the colcon package finder too

    pipeline_base_descr: "videotestsrc name=mysrc is-live=true ! videoconvert ! ximagesink"
    # pipeline_base_descr: "videotestsrc name=mysrc ! videoconvert ! rosimagesink name=mysink"
    # pipeline_base_descr: "v4l2src name=mysrc ! videoconvert ! ximagesink"
    # pipeline_base_descr: "filesrc location=tears-of-steel.mp4 ! decodebin ! videoconvert ! ximagesink"
    # pipeline_base_descr: "filesrc location=tears-of-steel.mp4 ! decodebin ! videoconvert ! rosimagesink name=mysink"

    ros_plugin_names:
      - "pause"
      - "framegate"
      #- "src_handler"
      #- "sink_handler"


    pause:
      type: "gst_pipes::gst_pipes_pause_srv"

    src_handler:
      type: "gst_pipes::gst_pipes_bridge"
      element_name: "mysrc"

    sink_handler:
      type: "gst_pipes::gst_pipes_bridge"
      element_name: "mysink"

    framegate:
      type: "gst_pipes::gst_pipes_framegate"
      element_name: "mysrc"
      pass_on_startup: true
